instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EvOfflineTest.kt

```diff
@@ -0,0 +1,130 @@
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.instrumentation_tests.utils.location.stayOnPosition
+import com.mapbox.navigation.instrumentation_tests.utils.routes.EvRoutesProvider
+import com.mapbox.navigation.instrumentation_tests.utils.routes.MockedEvRoutes
+import com.mapbox.navigation.instrumentation_tests.utils.tiles.OfflineRegions
+import com.mapbox.navigation.instrumentation_tests.utils.tiles.withMapboxNavigationAndOfflineTilesForRegion
+import com.mapbox.navigation.instrumentation_tests.utils.withMapboxNavigation
+import com.mapbox.navigation.instrumentation_tests.utils.withoutInternet
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.RouteRequestResult
+import com.mapbox.navigation.testing.ui.utils.coroutines.getSuccessfulResultOrThrowException
+import com.mapbox.navigation.testing.ui.utils.coroutines.requestRoutes
+import com.mapbox.navigation.testing.ui.utils.coroutines.routesUpdates
+import com.mapbox.navigation.testing.ui.utils.coroutines.sdkTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.setNavigationRoutesAsync
+import kotlinx.coroutines.flow.first
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+// TODO: remove in the scope of NAVAND-1351
+const val INCREASED_TIMEOUT_BECAUSE_OF_REAL_ROUTING_TILES_USAGE = 80_000L
+
+class EvOfflineTest : BaseCoreNoCleanUpTest() {
+
+    override fun setupMockLocation(): Location {
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            longitude = 13.361378213031003
+            latitude = 52.49813341962201
+        }
+    }
+
+    @Test
+    fun requestRouteWithoutInternetAndTiles() = sdkTest {
+        val testRoute = setupBerlinEvRoute()
+        withMapboxNavigation { navigation ->
+            withoutInternet {
+                val routes = navigation.requestRoutes(testRoute.routeOptions)
+                assertTrue(routes is RouteRequestResult.Failure)
+            }
+        }
+    }
+
+    @Test
+    fun requestOnlineRouteWithoutInternetHavingTiles() = sdkTest(
+        timeout = INCREASED_TIMEOUT_BECAUSE_OF_REAL_ROUTING_TILES_USAGE
+    ) {
+        val originalTestRoute = setupBerlinEvRoute()
+        withMapboxNavigationAndOfflineTilesForRegion(
+            OfflineRegions.Berlin
+        ) { navigation ->
+            navigation.startTripSession()
+            withoutInternet {
+                val requestResult = navigation.requestRoutes(originalTestRoute.routeOptions)
+                    .getSuccessfulResultOrThrowException()
+                assertEquals(RouterOrigin.Onboard, requestResult.routerOrigin)
+                navigation.setNavigationRoutesAsync(requestResult.routes)
+
+                assertEquals(
+                    "onboard router doesn't add charging waypoints",
+                    requestResult.routes.map { 2 },
```

Comment:
I don't want tests to fail in case onboard router starts returning 3 routes instead of two. For this specific case, the most important thing is that each of the routes have 2 waypoints.